addons/website_sale/models/product_template.py

```diff
@@ -168,11 +168,12 @@ def _get_sales_prices(self, pricelist):
         sales_prices = pricelist._get_products_price(self, 1.0)
         show_discount = pricelist.discount_policy == 'without_discount'
 
-        base_sales_prices = None
-        if show_discount:
-            base_sales_prices = self.price_compute(
-                'list_price', currency=pricelist.currency_id)
+        base_sales_prices = self.price_compute('list_price', currency=pricelist.currency_id)
 
+        # This is a structure that contains whether a product template has a pricelist for it
+        pricelist_contains_templates = {sale_price[0]: base_sale_price[1] != sale_price[1]
+                                        for sale_price, base_sale_price
+                                        in zip(sales_prices.items(), base_sales_prices.items())}
```

Comment:
This feels unsafe to me, I agree there is no reason that the order from `sales_price` differs from the order from `base_sales_prices` but I would still take it into account.

```suggestion
        base_sales_prices = self.price_compute('list_price', currency=pricelist.currency_id)

        # This is a structure that contains whether a product template has a pricelist for it
        pricelist_contains_templates = {product: price != base_sales_price[product]
                                        for product, price
                                        in sales_prices}
```

Reply:
Also side note I am not sure we actually need to create that dict, we could also just compare it when necessary (can be done once per template if you store it)